### ISSUE TITLE
Lag on display canvas html5

### ISSUE THREAD

**Issue by mataty501** [OwlCarousel2/OwlCarousel2#2413]
Hello, i'm trying to set some canvas circle percentage values over a basic owl caroussel, the problem is when i scroll there are some values whom don't display until they reach the first place on the caroussel, anyone knows how to fix this problem ?

i've already posted on stackoverflow, but nobody seems to know the reason.
https://stackoverflow.com/questions/51751978/html5-canvas-circle-lag-on-display

This is my code:
- https://codepen.io/m4t/pen/zLMZrB

Can somebody explain the problem, and how to fix it.

Thank you.

**Comment by taronarm** [comment]
Please find answer of your question and working example in stackoverflow: [https://stackoverflow.com/questions/51751978/html5-canvas-circle-lag-on-display/51773916#51773916](https://stackoverflow.com/questions/51751978/html5-canvas-circle-lag-on-display/51773916#51773916)

1. Owl carousel **creates clones of previous/next items** to allow group loop (e.g. than you clicked on navigation balls). As result you see clone of item, but during simple cloning of canvas element, the image will not be cloned. 
    
2. So you need to initialize canvas for each cloned element, but such as plugin doesn't allow you to get only cloned elements, you can filter them by class `cloned` or use flag for already initialized items (I prefer second way).
    
3. Never set an *"id"* attribute for each item of carousel, because you can get **duplicate of same `id`** in DOM (do to cloning).
    
4. It's good to **use same class for all items** and call canvas function only once, not for all items separately, because count of items can be dynamic.
